src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -910,6 +910,20 @@ trait Scanners extends ScannersCommon {
           setStrVal()
           token = STRINGLIT
         }
+      } else if (ch == '\\') {
```

Comment:
```suggestion
      } else if (ch == '\\' && !multiLine) {
```
then we can drop the `if `below